pallets/vested-rewards/src/lib.rs

```diff
@@ -120,6 +123,210 @@ pub struct CrowdloanUserInfo<AssetId> {
 pub use weights::WeightInfo;
 
 impl<T: Config> Pallet<T> {
+    #[cfg(feature = "wip")] // ORML multi asset vesting
+    fn do_claim_unlocked(
+        asset_id: AssetIdOf<T>,
+        who: &T::AccountId,
+    ) -> Result<BalanceOf<T>, DispatchError> {
+        let (locked, other_count): (Balance, u32) = Self::locked_balance(asset_id, who)?;
+        if locked.is_zero() && other_count.is_zero() {
+            // cleanup the storage and unlock the fund
+            <VestingSchedules<T>>::remove(who);
+            T::Currency::remove_lock(VESTING_LOCK_ID, asset_id, who)?;
+        } else if locked.is_zero() {
+            T::Currency::remove_lock(VESTING_LOCK_ID, asset_id, who)?;
+        } else {
+            T::Currency::set_lock(VESTING_LOCK_ID, asset_id, who, locked)?;
+        }
+        Ok(locked)
+    }
+
+    #[cfg(feature = "wip")] // ORML multi asset vesting
+    /// Returns locked balance based on current block number.
+    fn locked_balance(
+        asset_id: AssetIdOf<T>,
+        who: &T::AccountId,
+    ) -> Result<(BalanceOf<T>, u32), DispatchError> {
+        let now = frame_system::Pallet::<T>::current_block_number();
+        <VestingSchedules<T>>::mutate_exists(who, |maybe_schedules| {
```

Comment:
I think it's better to use `try_mutate_exists` here, because it may return error